packages/react-components/src/components/LocalVideoTile.tsx

```diff
@@ -54,17 +54,79 @@ export const LocalVideoTile = React.memo(
       localVideoSelectedDescription
     } = props;
 
+    const [localStreamRendererResult, setLocalStreamRendererResult] = React.useState<CreateVideoStreamViewResult>();
+
+    // When localVideoViewOptions change two things may happen:
+    //
+    // 1. Just the scaling mode has changed and we have access to the original LocalStreamRenderView
+    //   - In this case we just need to call updateScalingMode on the view and do not need to recreate the stream view
+    // 2. isMirrored has changed, or scalingMode change and we don't have access to the original LocalStreamRenderView
+    //   - In this case we need to dispose the old view and create a new one
+    //
+    // For scenario 1, we hold onto a ref to the scalingMode that persists across renders. When localVideoViewOptions.scalingMode
+    // differs from the persistent ref, we know to call updateScalingMode. We then subsequently update the ref to the new value for
+    // future renders.
+    //
+    // For scenario 2, we extract the localVideoViewOptions.isMirrored to ensure any change to the isMirrorred prop triggers the useEffect
+    // that recreates the local stream view. We must also here add an extra check for when the scaling mode changes and we do not have
+    // access to the original LocalStreamView (and hence cannot call updateScalingMode). When this happens we must also trigger the useEffect
+    // that recreates the local stream view.
+
+    const scalingModeRef = useRef(localVideoViewOptions?.scalingMode);
+    const newIsMirrored = localVideoViewOptions?.isMirrored;
+    const newScalingMode = localVideoViewOptions?.scalingMode;
+    const hasScalingModeChanged = scalingModeRef.current !== newScalingMode;
+    const updatingScalingModeDirectly = hasScalingModeChanged && !!localStreamRendererResult;
+
+    if (isAvailable && renderElement && newScalingMode && updatingScalingModeDirectly) {
+      localStreamRendererResult && localStreamRendererResult.view.updateScalingMode(newScalingMode);
+      scalingModeRef.current = localVideoViewOptions?.scalingMode;
+    }
+
+    // scalingModeForUseEffect will trigger the useEffect to recreate the local stream view only if the scaling mode has changed and
+    // we cannot call updateScalingMode on the view object directly. Otherwise it will be null to ensure the useEffect does not trigger
+    // when scaling mode changes.
+    const scalingModeForUseEffect = updatingScalingModeDirectly ? null : newScalingMode;
+
     useEffect(() => {
+      // Avoid race condition where onDisposeLocalStreamView is called before onCreateLocalStreamView
+      // and setLocalStreamRendererResult have completed
+      let wasLocalStreamDisposed = false;
+
       if (isAvailable && !renderElement) {
-        onCreateLocalStreamView?.(localVideoViewOptions);
+        (async (): Promise<void> => {
+          console.log('calling onCreateLocalStreamView');
+          const streamRendererResult = await onCreateLocalStreamView?.({
```

Comment:
TIL: JavaScript guarantees that an `async` function will run to the first `await` expression in its body synchronously: 
![image](https://user-images.githubusercontent.com/82062616/168339601-6236e90a-91c6-4d1e-a6b8-9b777d8b29da.png)

This is why you don't need to check for `wasLocalStreamDisposed` before calling `onCreateLocalStreamView`.

This need not be this way by the way. Other languages don't require concurrent stuff to being immediately (e.g. Go's goroutines can be scheduled at any time by the runtime, even if running on a single thread).